File: loan-repayment-java/src/test/java/com/loan/repayment/factory/ErrorResponseFactory.java
package com.loan.repayment.factory;

import com.loan.repayment.exception.ErrorResponse;

/**
 * 
 * @author Ramachandra
 *
 */
public class ErrorResponseFactory {

    /**
     * This method creates the error details for Nominal Rate Zero
     * 
     * @return ErrorResponse
     */
    public static ErrorResponse createErrorResponse() {
        //@formatter:off
		return ErrorResponse
				.builder()
				.errorDetails(ErrorDetailsFactory.createErrorDetails())
				.build();
		//@formatter:on
    }
}

File: loan-repayment-kotlin/src/test/kotlin/com/loan/repayment/factory/ErrorResponseFactory.kt
package com.loan.repayment.factory

import com.loan.repayment.exception.ErrorResponse

/**
 *
 * @author Rama chandra
 */
object ErrorResponseFactory {
	/**
	 * This method creates the error details for Nominal Rate Zero
	 *
	 * @return ErrorResponse
	 */
	fun createErrorResponse() : ErrorResponse {
		//@formatter:off
		return ErrorResponse(ErrorDetailsFactory.createErrorDetails())
		//@formatter:on
	}
}